bionic/executor.py

```diff
@@ -30,18 +32,39 @@ class AipExecutor:
 
     NON_ALPHANUMERIC_PATTERN = re.compile(r"\W")
 
-    def __init__(self, gcs_fs, aip_client, aip_config):
+    def __init__(
+        self,
+        gcs_fs,
+        aip_client,
+        aip_config: Config,
+        docker_build_enabled: bool,
```

Comment:
WDYT about replacing these two arguments with a single `docker_image_uri_func` argument, which is a function that returns the URI?